### PR TITLE
accounts/abi: implement new fallback functions

### PR DESCRIPTION
**Pull request by rjl493456442**
In Solidity v0.6.0, the original fallback is separated
into two different sub types: fallback and receive.

This PR addes the support for parsing new format abi
and the relevant abigen functionalities.

### CODE REVIEW

**Comment by rjl493456442** [comment]
@gballet It's ready to be reviewed, please check it if you are available :P

File: accounts/abi/method.go [hunk 1]
```diff
@@ -41,10 +41,23 @@ type Method struct {
...
+	// is generated by legacy compiler.
+	StateMutability string
+
+	// Legacy indicators generated by compiler before v0.6.0
+	Constant bool
+	Payable  bool
+
+	// The following two flags indicates whether the method is a
+	// special fallback introduced in solidity v0.6.0
+	Fallback bool
```
Comment (MariusVanDerWijden) [inline accounts/abi/method.go:58]: Maybe `IsFallback` is better, `abi.Fallback.Fallback` looks a bit weird.

**Comment by MariusVanDerWijden** [comment]
@gballet The ARM bug seems to already be in master, see https://travis-ci.org/github/ethereum/go-ethereum/jobs/674687814

**Comment by rjl493456442** [comment]
@MariusVanDerWijden I can check the arm testing failure. It's a part of les code. But I guess for this PR it's ok to ignore it.

**Comment by gballet** [comment]
@MariusVanDerWijden @rjl493456442 yes the bug is already in master, it got introduced by @zsfelfoldi's 0851646e48 change, which apparently got accidentally pushed into master last week. I am keeping this PR open a little bit longer because it's practical for me to investigate but I will revert my changes tomorrow at the latest and merge it.

File: les/utils/expiredvalue.go [hunk 2]
```diff
@@ -92,10 +93,13 @@ func (e *ExpiredValue) Add(amount int64, logOffset Fixed64) int64 {
 		e.Exp = integer
 	}
 	if base >= 0 || uint64(-base) <= e.Base {
+		fmt.Println("base=", base, uint64(-base), e.Base, uint64(base))
 		e.Base += uint64(base)
```
Comment (gballet) [inline les/utils/expiredvalue.go:97]: This is the line that is causing the problem. On arm64, `uint64(base) == 0` whereas on amd64 it's basically -1. I am trying to determine if the problem is with golang or if it's with arm64.

**Pull request merged**